DuckDuckGo/DuckPlayer/DuckPlayer.swift

```diff
@@ -112,73 +91,77 @@ struct InitialSetupSettings: Codable {
 public struct UserValues: Codable {
     enum CodingKeys: String, CodingKey {
         case duckPlayerMode = "privatePlayerMode"
-        case overlayInteracted
+        case askModeOverlayHidden = "overlayInteracted"
     }
     let duckPlayerMode: DuckPlayerMode
-    let overlayInteracted: Bool
+    let askModeOverlayHidden: Bool
 }
 
-final class DuckPlayer {
-    static let usesSimulatedRequests: Bool = {
-        if #available(macOS 12.0, *) {
-            return true
-        } else {
-            return false
+final class DuckPlayerSettings {
+    
+    public struct OriginDomains {
+        static let duckduckgo = "duckduckgo.com"
+        static let youtubeWWW = "www.youtube.com"
+        static let youtube = "youtube.com"
+        static let youtubeMobile = "m.youtube.com"
+    }
+    
+    var mode: DuckPlayerMode {
+        get {
+            AppDependencyProvider.shared.appSettings.duckPlayerMode
```

Comment:
Maybe we can inject a DependencyProvider with a default value of AppDependencyProvider.shared.appSettings? Just to make the injection a bit clear from the callsite and make testing easier if necessary

Reply:
Good call